checker/src/org/checkerframework/checker/upperbound/UpperBoundVisitor.java

```diff
@@ -0,0 +1,131 @@
+package org.checkerframework.checker.upperbound;
+
+import com.sun.source.tree.ArrayAccessTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import java.util.ArrayList;
+import java.util.List;
+import javax.lang.model.element.ExecutableElement;
+import org.checkerframework.checker.upperbound.qual.LTLengthOf;
+import org.checkerframework.checker.upperbound.qual.LTOMLengthOf;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.common.basetype.BaseTypeVisitor;
+import org.checkerframework.dataflow.analysis.FlowExpressions;
+import org.checkerframework.framework.source.Result;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.javacutil.TreeUtils;
+
+/** Warns about array accesses that could be too high. */
+public class UpperBoundVisitor extends BaseTypeVisitor<UpperBoundAnnotatedTypeFactory> {
+
+    protected final List<ExecutableElement> IndexFirstArgListMethods;
+
+    private static final String UPPER_BOUND = "array.access.unsafe.high";
+    private static final String UPPER_BOUND_LIST = "list.access.unsafe.high";
+
+    public UpperBoundVisitor(BaseTypeChecker checker) {
+        super(checker);
+        this.IndexFirstArgListMethods = new ArrayList<ExecutableElement>();
+        this.IndexFirstArgListMethods.add(
+                TreeUtils.getMethod(
+                        "java.util.List", "get", 1, checker.getProcessingEnvironment()));
+        this.IndexFirstArgListMethods.add(
+                TreeUtils.getMethod(
+                        "java.util.List", "set", 2, checker.getProcessingEnvironment()));
+        //// can't handle until TreeUtils.getMethod has a way to precisly handle method overloading
```

Comment:
I don't know. This would have to have come from Joe, though; he wrote this code. I think what he needs is a way to get only the version of list.remove that takes an index. I think the solution to this, though, isn't to implement that but rather to completely annotate the JDK, which solves this problem entirely.